src/TypeFormatter/LiteralUnionTypeFormatter.ts

```diff
@@ -43,8 +48,8 @@ export class LiteralUnionTypeFormatter implements SubTypeFormatter {
             };
         }
 
-        const values = uniqueArray(types.map(getLiteralValue));
-        const typeNames = uniqueArray(types.map(getLiteralType));
+        const values = uniqueArray(types.flatMap(getLiteralValues));
```

Comment:
this can probably become map again if we fix the thing below

Reply:
Did it for us since i also need this PR :)

351e144